ckanext/cfpb_extrafields/plugin.py

```diff
@@ -53,25 +183,96 @@ def update_package_schema(self):
+                                    tk.get_validator('ignore_missing')],
+            'pra_exclusion': [tk.get_converter('convert_from_extras'),
+                              tk.get_validator('ignore_missing')],
+            'dataset_last_modified_date': [tk.get_converter('convert_from_extras'),
+                                           v.reasonable_date_validator,
+                                           tk.get_validator('ignore_missing')],
+            'pra_omb_expiration_date': [tk.get_converter('convert_from_extras'),
+                                        tk.get_validator('ignore_missing')],
+            'privacy_has_privacy_act_statement': [tk.get_converter('convert_from_extras'),
+                                                  tk.get_validator('ignore_missing')],
+            'privacy_pia_notes': [tk.get_converter('convert_from_extras'),
+                                  tk.get_validator('ignore_missing')],
+            'records_retention_schedule': [tk.get_converter('convert_from_extras'),
+                                           tk.get_validator('ignore_missing')],
+            'procurement_document_id': [tk.get_converter('convert_from_extras'),
+                                        tk.get_validator('ignore_missing')],
+            'privacy_pia_title': [tk.get_converter('convert_from_extras'),
+                                  tk.get_validator('ignore_missing')],
+            'privacy_sorn_number': [tk.get_converter('convert_from_extras'),
+                                    tk.get_validator('ignore_missing')],
+            'dig_id': [tk.get_converter('convert_from_extras'),
+                       tk.get_validator('ignore_missing')],
+            'usage_restrictions' : [tk.get_converter('convert_from_extras'),
+                                    tk.get_validator('ignore_missing')],
         })
+        schema['resources'].update({
+                'approximate_total_size' : [ tk.get_validator('ignore_missing'),],
+                'content_temporal_range_end' : [v.reasonable_date_validator,  tk.get_validator('ignore_missing'),],
+                'content_temporal_range_start' : [v.reasonable_date_validator,  tk.get_validator('ignore_missing'),],
+                'intake_date' : [v.reasonable_date_validator,  tk.get_validator('ignore_missing'),],
+                'privacy_contains_pii' : [ tk.get_validator('ignore_missing'),],
+                'privacy_has_direct_identifiers' : [ tk.get_validator('ignore_missing'),],
+                'resource_type' : [ tk.get_validator('ignore_missing'),],
+                'sensitivity_level' : [ tk.get_validator('ignore_missing'),],
+                'storage_location' : [ tk.get_validator('ignore_missing'),],
+                'storage_location_path' : [ tk.get_validator('ignore_missing'),],  
+                'update_size' : [ tk.get_validator('ignore_missing'),],
+        })
+        schema['tags']['__extras'].append(tk.get_converter('free_tags_only'))
+        schema.update({
```

Comment:
Only because that's what's done in the docs http://docs.ckan.org/en/latest/extensions/adding-custom-fields.html#adding-tags-to-the-schema . There is a short explanation of the "vocabulary tags" (tags with a specified range of options that are labeled) being separated from the "free tags" (unconstrained tags) by the  schema['tags']['__extras'] line, but I don't have much understanding of that.